frontend/src/routes/UserManagement/Roles/Role/RolePermissions.tsx

```diff
@@ -1,18 +1,111 @@
 /* Copyright Contributors to the Open Cluster Management project */
-import { PageSection } from '@patternfly/react-core'
-import { useParams } from 'react-router-dom-v5-compat'
+import { Flex, FlexItem, Label, PageSection } from '@patternfly/react-core'
+
 import { useTranslation } from '../../../../lib/acm-i18next'
-import { AcmPage, AcmPageContent, AcmPageHeader } from '../../../../ui-components'
+import { AcmPage, AcmPageHeader, AcmTable, compareStrings, IAcmTableColumn } from '../../../../ui-components'
+
+import { PolicyRule } from '../../../../resources/kubernetes-client'
+import { useMemo, useCallback } from 'react'
+import { useCurrentRole } from '../RolesPage'
+import { cellWidth } from '@patternfly/react-table'
+
+export const kindToAbbreviation = (resourceOrKind: string): string => {
+  // Step 1: Extract the resource name part (before any path separators)
+  const resourceName = resourceOrKind.split(/[/\\?#]/)[0] || resourceOrKind
+
+  // Complex words: virtualmachines → VIRT, persistentvolumes → PERS
+  const abbrKind = resourceName.toUpperCase().slice(0, 4)
+  // Using logic in openshift console code
+  // https://github.com/openshift/console/blob/a228536605a2451468d172da9bb7b6ab812511ea/frontend/public/module/k8s/get-resources.ts#L29
+  const blocklist = ['ASS', 'FART']
+  return blocklist.includes(abbrKind) ? abbrKind.slice(0, -1) : abbrKind
```

Comment:
this is pretty personal. On one hand keeping `blokList` could improve readability, on the other hand there's no need to define a variable for a single shot usage, up to you, but I personally prefer to reduce boilerplate here, specially considering you added comments for the logic
```suggestion
  return ['ASS', 'FART'].includes(abbrKind) ? abbrKind.slice(0, -1) : abbrKind
```